src/__tests__/location.test.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { shallow,configure ,mount} from 'enzyme';
+import Enzyme from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+import Locations from "../components/Location/Locations";
+import { BrowserRouter as Router } from 'react-router-dom';
+
+// Configure Enzyme with React 16 adapter
+Enzyme.configure({ adapter: new Adapter() });
+
+describe('<Location/>', () => {
+
+
+    const locationlist= [{
+            id: 2,
+            Location: "Hyderabad",
+            Longitude: 78.4111,
+            Latitude: 17.4334
+        }
+        ];
+
+
+    const props = {
+        sort:true,
+        sortFn:jest.fn(),
+        updateToBrowser:jest.fn(),
+        LocationList:locationlist
+    };
+
+    it('should call sortFn when click on sortbyid button ', function () {
+        const component = shallow(<Locations {...props}/>);
+        const sortbyId = component.find('div div div button[id="sorting"]');
```

Comment:
This way of finding the component is tied up to the DOM hierarchy, so it is not future proof. This is better `component.find('#sorting')`

It can also be argued that why is `id` field responsible for testing. We generally avoid giving `id` fields for components. 
Change this to `<button testId="sortBtn"></button>` and `component.find('button[testId=sortBtn)`

testId makes the goal of the attribute more clearer.